### DISCUSSION TITLE
Is it possible to filter blog posts by status:all?

### DISCUSSION THREAD

**Discussion by t1t4nium** [datenstrom/community#878] [Ask a question] [answered]
Hello

I need to get all blog posts, regardless of their status (public, draft, unlisted, private, etc.).

This can be partially done by getting all posts by status using multiple calls:

```
<?php $unlisted = $this->yellow->content->index(true, true)->filter("status", "unlisted") ?>
<?php $public = $this->yellow->content->index(true, true)->filter("status", "public") ?>
<?php $draft = $this->yellow->content->index(true, true)->filter("status", "draft") ?>
<?php $private = $this->yellow->content->index(true, true)->filter("status", "private") ?>
```

But this make 4 lists. I need the same but in one list, sorted like any other list.

The purpose is to show a list of all posts in a private page.

I think the best and simple way is a solution where we can filter on a status named "all".

Something like that:

```
<?php $pages = $this->yellow->content->index(true, true)->filter("status", "all") ?>
```

Or is there any other way to do it?

Regards

**Comment by markseuffert** [comment, marked as answer]
The following gives you all pages with the page setting `Layout: blog`, regardless of their status:

```
<?php $pages = $this->yellow->content->index(true, true)->filter("layout", "blog") ?>
```

**Reply by t1t4nium** [reply]
Thank you

And with pagination

```
<?php $pages = $this->yellow->content->index(true, true)->filter("layout", "blog")->paginate($this->yellow->system->get("blogPaginationLimit")) ?>
```
